src/pproc/ecpoint.py

```diff
@@ -0,0 +1,469 @@
+                "typeOfGeneratingProcess": 13,
+                "typeOfPostProcessing": 206,
+                "indicatorOfUnitForTimeIncrement": 1,
+                "timeIncrement": 1,
+            }
+        )
+    return construct_message(template, grib_keys)
+
+
+def point_scale_template(
+    template: eccodes.GRIBMessage, pert_number: int, total_number: int, out_keys: dict
+) -> eccodes.GRIBMessage:
+    edition = out_keys.get("edition", template.get("edition"))
+    if edition not in (1, 2):
+        raise ValueError(f"Unsupported GRIB edition {edition}")
+
+    grib_keys = out_keys.copy()
+    if edition == 2:
+        grib_keys.update(
+            {
+                "productDefinitionTemplateNumber": 90,
+                "type": "pfc",
+                "inputProcessIdentifier": template.get("generatingProcessIdentifier"),
+                "typeOfPostProcessing": 206,
+                "indicatorOfUnitForTimeIncrement": 1,
+                "timeIncrement": 1,
+            }
+        )
+    return quantiles_template(template, pert_number, total_number, grib_keys)
+
+
+def compute_single_ens(
+    predictand: np.ndarray,
+    predictors: np.ndarray,
+    thr_inf: np.ndarray,
+    thr_sup: np.ndarray,
+    fer: np.ndarray,
+    codes_wt: np.ndarray,
+    wt_batch_size: int = 1,
+) -> tuple[np.ndarray, np.ndarray]:
+    num_fer = fer.shape[1]
+    num_wt = thr_inf.shape[0]
+    num_pred, num_gp = predictors.shape
+
+    pt_bc_allwt = np.zeros((num_fer, num_gp))
+    wt_allwt = np.zeros((num_gp,))
+    for index in range(0, num_wt, wt_batch_size):
+        end_index = min(index + wt_batch_size, num_wt)
+        logger.info(f"Weather types: {index} - {end_index - 1}")
+        wt_size = end_index - index
+
+        temp_wts = numexpr.evaluate(
+            "prod(where((predictors >= thr_inf) & (predictors < thr_sup), 1, 0), axis=1)",
+            local_dict={
+                "predictors": np.reshape(predictors, (1, num_pred, num_gp)),
+                "thr_inf": np.reshape(thr_inf[index:end_index], (wt_size, num_pred, 1)),
+                "thr_sup": np.reshape(thr_sup[index:end_index], (wt_size, num_pred, 1)),
+            },
+        )
+        temp_wts = np.where(np.any(np.isnan(predictors), axis=0), np.nan, temp_wts)
+
+        wt_allwt += np.einsum("i,i...", codes_wt[index:end_index], temp_wts)
+        wt_rain = numexpr.evaluate(
+            "pred * wts",
+            local_dict={
+                "pred": np.reshape(predictand, (1, num_gp)),
+                "wts": temp_wts,
+            },
+        )
+        cdf_wt = numexpr.evaluate(
+            "sum(wt_rain * (fer + 1), axis=0)",
+            local_dict={
+                "fer": np.reshape(fer[index:end_index], (wt_size, num_fer, 1)),
+                "wt_rain": np.reshape(wt_rain, (wt_size, 1, num_gp)),
+            },
+        )
+        numexpr.evaluate(
+            "pt + cdf", local_dict={"pt": pt_bc_allwt, "cdf": cdf_wt}, out=pt_bc_allwt
+        )
```

Comment:
Minor question: should we try to avoid using numexpr for trivial operations? (first and third, the second may still benefit)